### PR TITLE
discover: Implement Clone for Change

### PR DESCRIPTION
**Pull request by samvrlewis**
Implements Clone for discover::Change, if the underlying key and value both implement clone.

This is convenient for use-cases where a single change is duplicated so that it can be sent to multiple discover streams.

### CODE REVIEW

File: tower/src/discover/mod.rs [hunk 1]
```diff
@@ -104,3 +104,16 @@ pub enum Change<K, V> {
...
+    K: Clone,
+    V: Clone,
+{
+    fn clone(&self) -> Self {
+        match self {
+            Change::Insert(k, v) => Change::Insert(k.clone(), v.clone()),
+            Change::Remove(k) => Change::Remove(k.clone()),
+        }
+    }
+}
```
Comment (hawkw) [inline tower/src/discover/mod.rs:119]: This implementation seems like it could be derived?

**Review by hawkw — COMMENTED** [review]
I think implementing `Clone` for this seems like a reasonable addition to the API.

Is there a reason that `Clone` was manually implemented rather than derived?

File: tower/src/discover/mod.rs (hunk 1 above)
Comment (samvrlewis) [reply tower/src/discover/mod.rs]: Yep, you're right - I had some silly notions around deriving clone on enums with generics. Have fixed now, thanks!

**Review by hawkw — APPROVED** [review]
looks good to me, thanks!

**Pull request merged**